src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -73,15 +69,63 @@ public void storeMessage(Message message) {
       } catch (Exception e) {
         System.err.println("Error reading message.");
         System.err.println(entity.toString());
-        e.printStackTrace();
+        throw e;
       }
     }
 
     return messages;
   }
 
   /**
-   * Returns the total number of messages for all users. 
+   * Gets messages posted by a specific user.
+   *
+   * @return a list of messages posted by the user, or empty list if user has never posted a
+   *     message. List is sorted by time descending.
+   */
+  public List<Message> getMessages(String user) {
+    List<Message> messages = new ArrayList<>();
+
+    Query query =
+        new Query("Message")
+            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+            .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    try {
+      messages = loadMessages(results);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    return messages;
+  }
+
+  /**
+   * Gets messages posted by all users.
+   *
+   * @return a list of messages posted by all users, or empty list if all users have never posted a
+   *     message. List is sorted by time descending.
+   */
+  public List<Message> getMessages() {
+    List<Message> messages = new ArrayList<>();
+
+    Query query =
+        new Query("Message")
+            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
```

Comment:
What is the value of `user` here? I wouldn't think this would compile?

Reply:
Ah, yes! That's my bad. I'll get this fixed.

Reply:
Ah, yes! That's my bad. I just fixed it.